### DISCUSSION TITLE
Sorting and Filters (Settings)

### DISCUSSION THREAD

**Discussion by umbrellaplug** [umbrellaplug/umbrellaplug.github.io#83] [General]
Some new things have been added for sorting sources and filtering sources. There was so much added that I broke the settings out into their own category of Sorting and Filtering. I will give a breakdown of the settings.

### Top is "Sorting" options, there are only a few currently.
**Prioritize HDR and Dolby-Vision to top** - Makes sources with HDR or Dolby-Vision appear first in sources list. With this setting turned on you also have the ability to sort Dolby-Vision above HDR if you prefer those to be at the top.

**Prioritize AV1 to top** - Makes sources with AV1 type appear at the top of the list of sources.

**Prioritize HEVC to top** - Makes sources with HEVC type appear at the top the list of sources.

### Next we get to source filtering options. These are general options for filtering you can you for filtering results in sources to your preferences.

**Max Quality** - Max quality of video you want to see in your sources.

**Require Year in source for known Reboot shows** - Used to help filter down results better if you're searching for a show or movie that may be a remake or reboot. Adding this requires the year in the title also when gathering sources.

**Enable Movie File Size Limit** - Limit the size of the sources returned for movies.

**Enable Episode File Size Limit** - Limit the size of the sources returned for episodes of tv shows.

**Remove Duplicate Sources** - Removes sources that are duplicated due to the same file being found from multiple sources.

**Do not spam log with removed duplicates** - Avoids adding a log entry every time a duplicate source is removed.

**Do not show removed duplicates popup** - Avoids the pop-up saying "Duplicate Sources have been removed"

**Remove uncached torrents** - Removes uncached torrents from your sources list.

**Remove Dolby-Vision through Remove 3D** - All of these toggles can be turned on to remove these type of video codecs. For more explanations of file types and what is being removed please check the "help" under these settings in the addon.

### Source Filtering (Audio Codec) section is used for filtering sources based on audio preferences for codec.

**Remove OPUS through Remove MP3** - All of these toggles can be turned on to remove these types of audio codecs from sources returned. For more information on these please use the "help" button under these settings in the addon.

### Source Filtering (Audio Channel) - This section is used to filter sources based on the channel audio included with them.

**Remove 2 Channel through Remove 7 Channel** - These toggles are used to remove types of audio channels you may not want in your sources returned.

All of these settings are used to help filter down the types of results you get returned. They can be used with different types of devices and hardware so you can make sure you only get results in searches that are valid for the hardware you are running. Also very useful in helping to make sure Auto-Select gets only compatible results when turned on.

**Comment by LITUATUI** [comment]
Will it be possible to prioritize AV1 before HEVC?

This is what I would want:

1. AV1
2. HEVC (H.265)
3. AVC (H.264)
4. Others (DivX, Xvid, etc)

Thanks!

**Comment by adamosborne83** [comment]
> Will it be possible to prioritize AV1 before HEVC?
> 
> This is what I would want:
> 
> 1. AV1
> 2. HEVC (H.265)
> 3. AVC (H.264)
> 4. Others (DivX, Xvid, etc)
> 
> Thanks!

In source and filtering you can prioritise AV1 to top.

**Reply by LITUATUI** [reply]
I know, but I also want to give preference to HEVC against AVC.

**Reply by umbrellaplug** [reply]
Will look into this. I have a few updates in my list involving sources so I'll try to add it when I'm working on those files.

**Comment by XeonNAS** [comment]
Given I run a pi4 and can't for the life of me figure DV encoding out I am glad that I can prioritise hevc and drop DV results.
